### PR TITLE
crew: Add SHA512 and MD5 support

### PR DESCRIPTION
**Pull request by supechicken666**
Since many sources started to provide SHA512 sums, it could be convenient for packagers to copy it from sources  

Tested on `x86_64`

### CODE REVIEW

**Comment by saltedcoffii** [comment]
@supechicken666 could you also add Blake2b and MD5 support for Gentoo and Debian respectively?

File: crew [hunk 1]
```diff
@@ -916,7 +930,7 @@ def archive_package (pwd)
   Dir.chdir CREW_DEST_DIR do
     system "tar c#{@verbose}Jf #{pwd}/#{pkg_name} *"
   end
-  system "sha256sum #{pwd}/#{pkg_name} > #{pwd}/#{pkg_name}.sha256"
+  File.write("#{pwd}/#{pkg_name}.sha512", Digest::SHA512.hexdigest( File.binread("#{pwd}/#{pkg_name}") ) )
```
Comment (uberhacker) [inline crew:933]: Don't we want csum and csum_type here also?

File: crew [hunk 2]
```diff
@@ -529,17 +529,31 @@ def download
 
   uri = URI.parse url
   filename = File.basename(uri.path)
+  
   if source
-    sha256sum = @pkg.source_sha256
+    if @pkg.source_sha512
```
Comment (uberhacker) [inline crew:534]: Will this throw an error if source_sha512 is not set?

File: crew [hunk 3]
```diff
@@ -529,17 +529,31 @@ def download
...
+  elsif @pkg.binary_sha512
+    csum_type = 512
+    csum = @pkg.binary_sha512[@device[:architecture]]
+  elsif @pkg.binary_sha256
+    csum_type = 256
+    csum = @pkg.binary_sha256[@device[:architecture]]
   else
-    sha256sum = @pkg.binary_sha256[@device[:architecture]]
+    abort 'No checksum provided for this package, aborting...'.lightred
   end
```
Comment (uberhacker) [inline crew:549]: `case...when...else...end` would be more elegant.

File: crew (hunk 1 above)
Comment (supechicken666) [reply crew:933]: Not needed, this is the part used to generate the checksum for packaged binary

File: crew (hunk 2 above)
Comment (supechicken666) [reply crew:534]: No, as this is a instance variable (return nil if not set)

File: crew (hunk 3 above)
Comment (supechicken666) [reply crew:549]: Do you mean `case source...else...end`?

**Comment by supechicken666** [comment]
> @supechicken666 could you also add Blake2b and MD5 support for Gentoo and Debian respectively?

Sure, md5 support are also include in `digest` but blake2b require an extra [gem](https://github.com/mgomes/blake2b) to check

File: crew (hunk 3 above)
Comment (uberhacker) [reply crew:549]: You can use `when` instead of `elsif`.

File: crew (hunk 1 above)
Comment (uberhacker) [reply crew:933]: Not following.

**Comment by saltedcoffii** [comment]
Package MD5 for now and I'll package Blake2B and add support for it later

File: crew (hunk 1 above)
Comment (supechicken666) [reply crew:933]: I mean this is a part only used in `crew build`, so it is fine to default `sha512sum`

File: crew (hunk 3 above)
Comment (supechicken666) [reply crew:550]: Done

File: crew (hunk 1 above)
Comment (uberhacker) [reply crew:939]: If you are going to do this then you might as well get rid of the other options for binary checksums and just go with sha512sum.

Comment (supechicken) [reply crew:939]: Can we keep the compatibility for them? (since many source are still providing `sha256sum` in their checksum file)

Comment (uberhacker) [reply crew:939]: Sure but the way you have it set up currently, there is no way to implement md5 or blake2b checksums.

Comment (supechicken) [reply crew:935]: Removed md5 support

**Pull request closed without merging**